components/menu/interface.ts

```diff
@@ -5,7 +5,11 @@ import type {
   SubMenuType as RcSubMenuType,
 } from 'rc-menu/lib/interface';
 
-export interface MenuItemType extends RcMenuItemType {
+export type DataAttributes = {
+  [Key in `data-${string}`]: string | number;
```

Comment:
Either, the value should be `any` or just `string`.

Let me evaluate:
`any`: Anything is allowed as a `data-` attribute value.
`string`: Any value, except for `null | undefined` will be a string in the DOM node. `null` and `undefined` values result in the attribute to not be present at all. As everything is being `toString`ed, array values will be comma separated and objects will become the good old `[object Object]`.

I think, `any` is the way to go here, as everything should be accepted at this place.